File: ObjectiveC/FlexGrid101/FlexGrid101/FilterFormController.m
//
//  FilterFormController.m
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

#import "FilterFormController.h"
#import "CustomerData.h"
#import "FilterController.h"
#import "FlexGridKit/FlexGridKit.h"
#import "SharedFilterData.h"

@interface FilterFormController (){
    NSMutableArray *pickerData;
    UITextField *_filterField;
    SharedFilterData *sharedData;
}

@end

@implementation FilterFormController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    sharedData = [SharedFilterData sharedInstance];
    pickerData =[[NSMutableArray alloc] initWithObjects:@"Contains", @"Begins with", @"Ends with", @"Equals", nil];
    UIPickerView *filterPicker = [[UIPickerView alloc] init];
    
    filterPicker.delegate = self;
    filterPicker.showsSelectionIndicator = YES;
    filterPicker.hidden = false;
    [filterPicker selectRow:0 inComponent:0 animated:false];

    _filterField = [[UITextField alloc] init];
    _filterField.delegate = self;
    _filterField.text = @"Enter text to Filter";
    _filterField.returnKeyType = UIReturnKeyDone;
    _filterField.keyboardType = UIKeyboardTypeDefault;
    _filterField.backgroundColor = [UIColor lightGrayColor];
    
    
    UIButton *filterButton = [UIButton buttonWithType:UIButtonTypeSystem];
    
    [filterButton setTitle:@"Filter" forState:UIControlStateNormal];
    [filterButton.titleLabel setTextAlignment:NSTextAlignmentCenter];
    [filterButton addTarget:self action:@selector(filterButtonClicked) forControlEvents:UIControlEventTouchUpInside];
    [filterButton setBackgroundColor:[UIColor blueColor]];
    filterButton.tintColor = [UIColor whiteColor];
    
    filterPicker.tag = 1;
    filterButton.tag = 2;
    
    [self.view addSubview:_filterField];
    [self.view addSubview:filterPicker];
    [self.view addSubview:filterButton];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}
- (void)viewDidLayoutSubviews{
    [super viewDidLayoutSubviews];
    UIPickerView *filterPicker = (UIPickerView*)[self.view viewWithTag:1];
    UIButton *filterButton = (UIButton*)[self.view viewWithTag:2];
    _filterField.frame = CGRectMake(self.view.bounds.size.width/4, 65, self.view.bounds.size.width/2, 50);
    filterPicker.frame = CGRectMake(self.view.bounds.size.width/4, 100, self.view.bounds.size.width/2, 162);
    filterButton.frame = CGRectMake(self.view.bounds.size.width/4, 300, self.view.bounds.size.width/2, 50);
}
-(void)filterButtonClicked{
    sharedData.filterSet = true;
    [self.navigationController popViewControllerAnimated:true];
}

- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView{
    return 1;
}

- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component{
    return [pickerData count];
}

- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component{
    sharedData.filterOperation = @(row);
}

- (NSString*)pickerView:(UIPickerView *)pickerView titleForRow:(NSInteger)row forComponent:(NSInteger)component{
    return [pickerData objectAtIndex:row];
}

-(BOOL)textFieldShouldBeginEditing:(UITextField *)textField{
    return true;
}

-(void)textFieldDidBeginEditing:(UITextField *)textField{
    [textField selectAll:nil];
}

-(BOOL)textFieldShouldEndEditing:(UITextField *)textField{
    return true;
}

-(void)textFieldDidEndEditing:(UITextField *)textField{
    sharedData.filterString = textField.text;
}

-(BOOL)textFieldShouldReturn:(UITextField *)textField{
    [textField resignFirstResponder];
    return true;
}

@end

File: Swift/FlexGrid101/FlexGrid101/FilterFormController.swift
//
//  FilterFormController.swift
//  FlexGrid101
//
//  Copyright (c) 2015 GrapeCity. All rights reserved.
//

import UIKit

class FilterFormController: UIViewController, UIPickerViewDataSource, UIPickerViewDelegate, UITextFieldDelegate {
    
    var _pickerData = NSMutableArray()
    var _filterField = UITextField()
    var _filterPicker = UIPickerView()
    var _filterButton = UIButton()
    
    let sharedData = SharedFilterData.sharedInstance
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        checkStartingFilterValues()

        // Do any additional setup after loading the view.
        _pickerData = [NSLocalizedString("Contains", comment: ""), NSLocalizedString("Begins with", comment: ""), NSLocalizedString("Ends with", comment: ""), NSLocalizedString("Equals", comment: "")]
        
        _filterPicker.delegate = self
        _filterPicker.showsSelectionIndicator = true
        _filterPicker.hidden = false
        
        _filterButton = UIButton(type: UIButtonType.System)
        _filterButton.setTitle(NSLocalizedString("Filter", comment: ""), forState: UIControlState.Normal)
        _filterButton.addTarget(self, action: "filterButtonClicked", forControlEvents: UIControlEvents.TouchUpInside)
        _filterButton.titleLabel?.textAlignment = NSTextAlignment.Center
        _filterButton.backgroundColor = UIColor.blueColor()
        _filterButton.tintColor = UIColor.whiteColor()
        
        _filterField.delegate = self
        _filterField.returnKeyType = UIReturnKeyType.Done
        _filterField.keyboardType = UIKeyboardType.Default
        _filterField.backgroundColor = UIColor.lightGrayColor()

        self.view.addSubview(_filterField)
        self.view.addSubview(_filterPicker)
        self.view.addSubview(_filterButton)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func viewDidLayoutSubviews() {
        super.viewDidLayoutSubviews()
        _filterField.frame = CGRectMake(self.view.bounds.size.width/4, 65, self.view.bounds.size.width/2, 50)
        _filterPicker.selectRow(self.sharedData.filterOperation, inComponent: 0, animated: false)
        _filterPicker.frame = CGRectMake(self.view.bounds.size.width/2 - 160, 100, 320, 162)
        _filterButton.frame = CGRectMake(self.view.bounds.size.width/4, 300, self.view.bounds.size.width/2, 50)
    }
    
    
    override func viewDidAppear(animated: Bool) {
        super.viewDidAppear(true)

    }
    
    func checkStartingFilterValues(){
        
        if(self.sharedData.filterOperation == nil){
            self.sharedData.filterOperation = 0;
        }
        if(self.sharedData.filterString != nil && self.sharedData.filterString != ""){
            _filterField.text = self.sharedData.filterString
        }
        else{
            _filterField.text = NSLocalizedString("Enter text to Filter",comment:"")
        }
    }
    func filterButtonClicked(){
        self.sharedData.filterSet = true
        self.sharedData.filterString = _filterField.text
        navigationController?.popViewControllerAnimated(true)
    }

    func numberOfComponentsInPickerView(pickerView: UIPickerView) -> Int {
        return 1
    }
    func pickerView(pickerView: UIPickerView, numberOfRowsInComponent component: Int) -> Int {
        return _pickerData.count
    }
    func pickerView(pickerView: UIPickerView, didSelectRow row: Int, inComponent component: Int) {
        self.sharedData.filterOperation = row
    }
    func pickerView(pickerView: UIPickerView, titleForRow row: Int, forComponent component: Int) -> String! {
        return _pickerData.objectAtIndex(row) as! String
    }
    func textFieldShouldBeginEditing(textField: UITextField) -> Bool {
        return true
    }
    func textFieldDidBeginEditing(textField: UITextField) {
        textField.selectAll(nil)
    }
    func textFieldShouldEndEditing(textField: UITextField) -> Bool {
        return true
    }
    func textFieldDidEndEditing(textField: UITextField) {
        self.sharedData.filterString = textField.text
    }
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        return true
    }    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
